feat(hardware): update motor position estimation from hardware controller

fsinapi opened the pull request:
# Overview
This PR adds a _private_ function in OT3API to enable users to send the command to reset the position estimation for an axis.

# Changelog
- Added an API command to update the position estimation for a set of axes
- Added real & simulated backend support to update position estimation

# Review requests

- [x] Does this make sense to expose on this level, or should this only be available internally to the hardware controller?
- [x] Test that using the API function on a robot sends the command to the correct nodes

# Risk assessment

Low, this is new functionality not integrated in any existing protocols.

fsinapi committed:
Added command to update position estimation from hardware controller

fsinapi committed:
Updated implementation of send/receive for motor position status

fsinapi committed:
Added a test for motor position estimation update parser

fsinapi committed:
Added simulator implementation to update motor estimation
- Removed debugging print statement
- Added a test for ot3controller and ot3api

sfoster1 reviewed (changes requested):
I think this doesn't want to be external but since it's python we can just make it "private" by prefixing its name with an underscore.

I also think it should only let you do this for nodes that actually have valid encoder estimates, and we should have a method the hardware controller can use to inspect the motion status of all the nodes.

fsinapi committed:
Merge branch 'edge' into RLIQ-288_update_estimation_hwc

fsinapi committed:
Addressed review comments:
- Made the API function private
- Added check to ensure all axes having position updated also have a valid encoder position

sfoster1 reviewed (approved):
Nice! Looks good!
